### PR TITLE
Warn about URLs with non-latin chars to prevent phishing attempts

### PR DESCRIPTION
**Pull request by imikailoby**
Option `ascii_only` (default: false) in the `urls_to_links` method has been added, which allows checking references for the presence of non-latin characters. If such characters are found, the link will not be displayed like html link (`<a href='...'>...</a>`), instead of it, a warning message will be returned wrapped in a `<b/>` tag.
Tests for validation was written.

**Examples:**

```
1. тест [WARNING, URL CONTAINS NON-LATIN LETTERS: http://vk.сom]
2. test https://vk.com
3. тест http://vk.com and test [WARNING, URL CONTAINS NON-LATIN LETTERS: http://vk.cоm]
```

**Live examples:**
live example

P.s. don't forget update [rubydoc](https://www.rubydoc.info/gems/string_master)

### CODE REVIEW

File: lib/string_master/string_master.rb [hunk 1]
```diff
@@ -67,10 +67,17 @@ def urls_to_images(options = {})
   def urls_to_links(options = {})
     wrap_with = options[:wrap_with] || ['','']
     html_options = options[:html_options] || ''
-    @modified_string.gsub!(
-      /(\s|^|\A|\n|\t|\r)((http|https):\/\/.*?)([,.])?(\s|$|\n|\Z|\t|\r|<)/,
-      '\1<a href="\2" ' + html_options + '>\2</a>\4\5'
-    )
+    validate = options[:ascii_only] || false
```
Comment (snitko) [inline lib/string_master/string_master.rb]: Variable name here is non-description. Validate what? Same for method name.

File: lib/string_master/string_master.rb [hunk 2]
```diff
@@ -67,10 +67,17 @@ def urls_to_images(options = {})
...
+
+    # ascii_only option - show URLs with warning about contain non-latin characters
+    if validate == true
```
Comment (snitko) [inline lib/string_master/string_master.rb]: Why use an additional variable `validate` at all? Why not `if options[:ascii_only]`?

Comment (imikailoby) [reply lib/string_master/string_master.rb]: Oops. Fixed

File: lib/string_master/string_master.rb (hunk 1 above)
Comment (imikailoby) [reply lib/string_master/string_master.rb]: Comments has been added above `if options[:ascii_only]` conditions and above `validate_url` method.

**Pull request merged**